src/app/features/home/custom-camera/custom-camera.page.ts

```diff
@@ -289,14 +296,59 @@ export class CustomCameraPage implements OnInit, OnDestroy {
     }
   }
 
-  async confirmCurrentCapture() {
-    if (this.curCaptureFilePath && this.curCaptureType) {
+  async confirmCurrentCapture(): Promise<void> {
+    try {
+      const shouldAskSaveToCameraRoll =
+        await this.customCameraService.shouldAskSaveToCameraRoll();
+      if (shouldAskSaveToCameraRoll === false) {
+        await this.uploadCurrentCapture();
+        return;
+      }
+
+      const confirmed = await this.showSaveToCameraRollConfirmAlert();
+      if (!confirmed) {
+        await this.customCameraService.setShouldSaveToCameraRoll('no');
+        await this.uploadCurrentCapture();
+        return;
+      }
+
+      await this.customCameraService.setShouldSaveToCameraRoll('yes');
+      await this.uploadCurrentCapture();
+    } catch (error: unknown) {
+      this.errorService.toastError$(error).subscribe();
+    }
+  }
+
+  private async showSaveToCameraRollConfirmAlert(): Promise<boolean> {
+    const header = this.translocoService.translate(
+      'customCamera.saveToDeviceGalleryAlert.header'
+    );
+    const message = this.translocoService.translate(
+      'customCamera.saveToDeviceGalleryAlert.messsage'
+    );
+    return await this.confirmAlert.present({ header, message });
+  }
+
+  private async uploadCurrentCapture() {
+    try {
+      if (!this.curCaptureFilePath || !this.curCaptureType) return;
+
+      const should = await this.customCameraService.getShouldSaveToCameraRoll();
+      if (should === 'yes') {
+        this.customCameraService.saveCaptureToUserDevice(
+          this.curCaptureFilePath
+        );
+      }
+
       this.customCameraService.uploadToCapture(
         this.curCaptureFilePath,
         this.curCaptureType,
         this.curCaptureCameraSource
       );
+
       this.leaveCustomCamera();
+    } catch (error: unknown) {
+      this.errorService.toastError$(error).pipe(take(1)).subscribe();
     }
   }
 
```

Comment:
The changes you made in the `CustomCameraPage` add the functionality to prompt the user with a confirmation alert to save the capture to their device gallery. The code seems well-implemented and the additional methods are easy to understand.

Here are a few suggestions to improve the code:

1.  Use `async/await` in `uploadCurrentCapture`:

In the `uploadCurrentCapture` method, you are using the `then` syntax with a `Promise`. Instead, you can use `async/await` for better readability:

```typescript

private async uploadCurrentCapture() {
  try {
    if (!this.curCaptureFilePath || !this.curCaptureType) return;

    const should = await this.customCameraService.getShouldSaveToCameraRoll();
    if (should === 'yes') {
      await this.customCameraService.saveCaptureToUserDevice(this.curCaptureFilePath);
    }

    await this.customCameraService.uploadToCapture(
      this.curCaptureFilePath,
      this.curCaptureType,
      this.curCaptureCameraSource
    );

    this.leaveCustomCamera();
  } catch (error: unknown) {
    this.errorService.toastError$(error).pipe(take(1)).subscribe();
  }
}
```

1.  Add comments to new methods:

Adding comments to the new methods like `confirmCurrentCapture`, `showSaveToCameraRollConfirmAlert`, and `uploadCurrentCapture` will make the code more readable and maintainable.

Overall, the changes made to the `CustomCameraPage` look good and the added functionality appears to be well-implemented.